docs/api-reference/next/link.md

```diff
@@ -218,3 +218,37 @@ The default behavior of `Link` is to scroll to the top of the page. When there i
   Disables scrolling to the top
 </Link>
 ```
+
+## With Next.js 13 Middleware
+
+It's common to use [Middleware](/docs/advanced-features/middleware) for authentication or other purposes that involve redirecting the user to a different page. In versions prior to Next.js 13, the redirect is handled on the client instead of the server. So, in order for the `<Link />` component to properly prefetch links with redirects via Middleware, you need to tell Next.js both the URL to display and the URL to prefetch.
```

Comment:
```suggestion
It's common to use [Middleware](/docs/advanced-features/middleware) for authentication or other purposes that involve rewriting the user to a different page. In order for the `<Link />` component to properly prefetch links with rewrites via Middleware, you need to tell Next.js both the URL to display and the URL to prefetch. This is required to avoid un-necessary fetches to middleware to know the correct route to prefetch. 
```